src/runtime/nitro/plugins/50-updateCsp.ts

```diff
@@ -1,12 +1,19 @@
 import { defineNitroPlugin } from '#imports'
 import { resolveSecurityRules } from '../context'
 import type { ContentSecurityPolicyValue } from '../../../types/headers'
+import { useLogger } from '@nuxt/kit'
 
 /**
  * This plugin updates the CSP directives with the nonce and hashes generated by the server.
  */
 export default defineNitroPlugin((nitroApp) => {
   nitroApp.hooks.hook('render:html', (response, { event }) => {
+    if (response.island) {
```

Comment:
Where does `reponse.island` comes from ?

Reply:
It is provided by Nuxt:

https://github.com/nuxt/nuxt/blob/c0fb30d6d0f06a0c1160e20c9c8f819075d62b14/packages/nuxt/src/core/runtime/nitro/renderer.ts#L503

Reply:
ah yes 🤦 confused it with the `render:response` hook